solutions/swb-reference/src/environment/ec2VSCode1710/ec2VSCode1710TerminateSSM.yaml

```diff
@@ -0,0 +1,90 @@
+description: SSM document to terminate a Vscode instance
+schemaVersion: '0.3'
+parameters:
+  ProvisionedProductId:
+    type: String
+    description: 'The ID of the Vscode provisioned product being terminated.'
+    allowedPattern: ^pp-[a-zA-Z0-9_\-]+$
+  TerminateToken:
+    type: String
+    description: '(Required) The unique token ID assigned to this request.'
+    allowedPattern: ^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$
+  EnvId:
+    type: String
+    description: 'The ID of the environment resource stored in DB'
+    allowedPattern: ^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$
+mainSteps:
+  - name: Terminate
+    action: 'aws:executeAwsApi'
+    maxAttempts: 3
+    timeoutSeconds: 30
+    inputs:
+      Service: servicecatalog
+      Api: TerminateProvisionedProduct
+      ProvisionedProductId: '{{ ProvisionedProductId }}'
+      TerminateToken: '{{ TerminateToken }}'
+    outputs:
+      - Name: RecordId
+        Type: String
+        Selector: '$.RecordDetail.RecordId'
+    nextStep: WaitForTerminating
+  - name: WaitForTerminating
+    action: 'aws:waitForAwsResourceProperty'
+    timeoutSeconds: 600
```

Comment:
Added a comment above

Reply:
made the suggested changes in commit c3ac4a835b3b6886d84d5c843038e0f81376215b